pkg/registry/maven/maven.go

```diff
@@ -170,6 +170,6 @@ func (r HTTPRegistry) ReleaseURL(ctx context.Context, pkg, version, typ string)
 		return "", errors.New("package identifier not of form 'group:artifact'")
 	}
 	artifactPath := path.Join(strings.ReplaceAll(g, ".", "/"), a, version, fmt.Sprintf("%s-%s%s", a, version, typ))
-	artifactURL := releaseURL.ResolveReference(urlx.MustParse(artifactPath))
-	return artifactURL.String(), nil
+	artifactURL := releaseURL.JoinPath(artifactPath).String()
```

Comment:
nit: i think the idea of having the "artifactURL" variable name was that it was of URL type. stringifying on this line isn't wrong but I might prefer keeping the String() where it was

Reply:
Makes sense. I will fix.